core/chains/evm/chain.go

```diff
@@ -96,14 +96,11 @@ type LegacyChainContainer interface {
 
 var _ LegacyChainContainer = &LegacyChains{}
 
-func NewLegacyChains(cfg AppConfig, m map[string]Chain) (*LegacyChains, error) {
-	if cfg == nil {
-		return nil, fmt.Errorf("must provide non-nil app config")
-	}
```

Comment:
I was going to wrap this in a `Must*` variant to remove the `err` var up the stack that was being shadowed in the sub-tests, but I think it's simpler to avoid the nil possibility all together. The callers can still check explicitly if necessary.
@krehermann wdyt?

Reply:
yeah, this is better. not sure why i passed in AppConfig, probably was confused and thought i needed more of it